### PR TITLE
[Docs] Add auth modes and dark site info to application

### PR DESCRIPTION
**Pull request by TomerShor**
Explain how to set the different authentication modes, and what is required for deploying an application runtime in a dark (air-gapped) environment.

https://iguazio.atlassian.net/browse/ML-7390

### CODE REVIEW

File: docs/runtimes/application.ipynb [hunk 1]
```diff
@@ -92,6 +92,74 @@
    "metadata": {
```
Comment (alonmr) [inline docs/runtimes/application.ipynb]: <p>Specifying "typically none, unless" emphasises open source over enterprise which IMO is wrong. I would say something like "is none for open source and access-key for Iguazio..."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/6176/discussion/#comment-1720957536'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (alonmr) [inline docs/runtimes/application.ipynb]: <pre class="ql-syntax" spellcheck="false">Line #8.    # for Basic authentication mode, credentials are also required.
</pre><p> </p><p>some inconsistencies here with capital letters not clear why</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/6176/discussion/#comment-1720957537'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (alonmr) [inline docs/runtimes/application.ipynb]: <pre class="ql-syntax" spellcheck="false">Line #2.    import mlrun.runtimes.application
</pre><p>this import should also work</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/6176/discussion/#comment-1720957538'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (TomerShor) [reply docs/runtimes/application.ipynb]: actually, this is better:
```
import mlrun.runtimes

# and use:
mlrun.runtimes.ApplicationRuntime
```

**Review by alonmr — APPROVED** [review]

**Pull request merged**